Chapter6/reliable_fsm.erl

```diff
@@ -0,0 +1,120 @@
+-module(reliable_fsm).
+-export([start/0, stop/0, un_hook/0, hook/0, other_un_hook/0, other_hook/0, call/1, get_call/1]).
+-export([idle/0, dial/0, wait_answer/1, ringing/1, connected/1]).
+
+start() -> register(cell, spawn(reliable_fsm, idle, [])).
+stop() 	-> cell ! stop.
+
+un_hook() 			-> msg(off_hook).
+hook() 				-> msg(on_hook).
+other_un_hook() 	-> msg(other_off_hook).
+other_hook() 		-> msg(other_on_hook).
+call(Number) 		-> msg({Number, calling}).
+get_call(Number) 	-> msg({Number, incoming}).
+
+
+% Exercise 6-2: A Reliable Mutex Semaphore
```

Comment:
There's mistake in book which points to wrong page. It must be p129. I think idea of ex 6.2 to make reliable mutex, which won't be locked forever when processes which asked for it dies unexpectedly.
Could you please implement it?

Reply:
It happens because your spawned processes terminate their execution right after they send wait message to mutex. Mutex, in turn, links itself to new processes and trap exit signals. That's why it receives 'EXIT' message when your first spawned processes dies normally and terminates itself.
Second processes tries to send wait to mutex, but since it's already dead you got that badarg  error.
Mutex should not be terminated if process, that acquired lock dies. It should go to `free` state and wait for new lock requests.

Reply:
Just updated the reliable_mutex module, its working with both the link and reference method, I've also added a test/0 function to show it working